### PR TITLE
px_uploader: Added progress bar for erasing and uploading firmware

### PR DESCRIPTION
**Pull request by Zefz**
Minor change to the px_uploader tool that adds progress bars to erasing and upload progress. Makes it easier to see that the upload process is not hanging and how much is remaining.

Example:
```
sn: 001b00293532470136343032
erase...
progress:[====================] 100.00%
erase complete!
program...
progress:[======              ] 35.12%
```

This has been verified to work correctly on 2 Linux systems and 1 OSX system.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Awesome! I don't think VT100 dependency is an issue, so we're definitely going for it. The win tool chain also supports VT100

File: Tools/px_uploader.py [hunk 1]
```diff
@@ -478,7 +508,7 @@ def send_reboot(self):
         for port in portlist:
 
                 #print("Trying %s" % port)
-
+
```
Comment (LorenzMeier) [inline Tools/px_uploader.py]: Whitespace :)

**Comment by Zefz** [comment]
@LorenzMeier According to this comment the read speed must be < 64 or else something overflows. Is this also true for writing? And is still a real issue? (programming bigger blocks of data seems to work fine for me) https://github.com/Zefz/Firmware/blob/px_uploader_improvements/Tools/px_uploader.py#L172

**Comment by LorenzMeier** [comment]
I would guess that larger blocks work now as well. But I don't want to update that before we haven't run a little more testing on it.

**Pull request merged**